src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -1503,6 +1623,10 @@ METHOD(kernel_ipsec_t, add_sa, status_t,
 
 	LOG(DBG_KNL, SL_LOG_INFO, "successfully added SA " ADD_SA_FMT,
 		ADD_SA_FMT_ARGS(id, data));
+
+	ike_sa_t* ike_sa = data->child_sa->get_ike_sa(data->child_sa);
```

Comment:
Same question regarding ike_rekey - Are you sure that on child rekey the ipsec sa lifetime needs to be reset?